_search-plugins/search-relevance/rerank-by-field.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: default
+title: Reranking by a field
+parent: Reranking search results
+grand_parent: Search relevance
+has_children: false
+nav_order: 20
+---
+
+# Reranking search results by a field
+Introduced 2.18
+{: .label .label-purple }
+
+You can use a `by_field` rerank type to rerank search results by a document field. Reranking search results by a field is useful if a model has already run and produced a numerical score for your documents or if a previous search response processor was applied and you want to rerank documents differently based on an aggregated field.
+
+To implement reranking, you need to configure a [search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/) that runs at search time. The search pipeline intercepts search results and applies the [`rerank` processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rerank-processor/) to them. The `rerank` processor evaluates the search results and sorts them based on the new scores obtained from a document field. 
+
+## Running a search with reranking
+
+To run a search with reranking, follow these steps:
+
+1. [Configure a search pipeline](#step-1-configure-a-search-pipeline).
+1. [Create an index for ingestion](#step-2-create-an-index-for-ingestion).
+1. [Ingest documents into the index](#step-3-ingest-documents-into-the-index).
+1. [Search using reranking](#step-4-search-using-reranking).
+
+## Step 1: Configure a search pipeline
+
+Configure a search pipeline with a [`rerank` processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rerank-processor/) and specify the `by_field` rerank type. The pipeline sorts by the `reviews.stars` field (specified by a complete dot path to the field) and returns the original query scores for all documents along with their new scores:
+
+```json
+PUT /_search/pipeline/rerank_byfield_pipeline
+{
+  "response_processors": [
+    {
+      "rerank": {
+        "by_field": {
+          "target_field": "reviews.stars",
+          "keep_previous_score" : true
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+For more information about the request fields, see [Request fields]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/rerank-processor/#request-body-fields).
+
+## Step 2: Create an index for ingestion
+
+In order to use the `rerank` processor defined in your pipeline, create an OpenSearch index and add the pipeline created in the previous step as the default pipeline:
+
+```json
+PUT /book-index
+{
+  "settings": {
+    "index.search.default_pipeline" : "rerank_byfield_pipeline"
+  },
+  "mappings": {
+    "properties": {
+      "title": {
+        "type": "text"
+      },
+      "author": {
+        "type": "text"
+      },
+      "genre": {
+        "type": "keyword"
+      },
+      "reviews": {
```

Comment:
@brianf-aws: Does the `reviews` field need to be of an `object` type?

Reply:
if reviews points to a nuermical score it can be used for example

```
{
   reviews: 5
}
```
then the targetField would be just reviews. The target field depends on the knowledge of the user really and how well they know their documents